### PR TITLE
Paper: Mamba Models a replacement for Transformers?

### PR DESCRIPTION
**Pull request by JaynouOliver**
If you are creating this PR in order to submit a draft of your paper, please name your PR with `Paper: <title>`. An editor will then add a `paper` label and GitHub Actions will be run to check and build your paper.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#preview-your-paper) for more information.

_Editor_: Meghann Agarwal @mepa

_Reviewers_: 
- Tetsuo Koyama @tkoyama010
- Rohit Goswami @HaoZeke

### CODE REVIEW

**Comment by JaynouOliver** [comment]
@hongsupshin is there any specific mistake from our side which led to curvenote checks getting skipped? Can't see the deployment preview.

**Comment by tkoyama010** [comment]
Hi. Thanks for invitation. I will review this paper :)

**Comment by HaoZeke** [comment]
As per the email received yesterday from the Proc. Comm. I am happy to review this submission.

**Comment by JaynouOliver** [comment]
Hi, @tkoyama010 @HaoZeke just wanted to know if there are any updates regarding the review, if I do have to make any changes from here.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you for the great paper. State-space models are something I have used in my own career and I am amazed that they can be applied in this way. Is it possible to share the URL of the GitHub repository you used for this model in this paper? I am sure it would be even better if you could share it so we can actually implement this idea.

**Comment by JaynouOliver** [comment]
I can share the code written originally by the authors of MAMBA https://github.com/state-spaces/mamba
Just wanted to check with you before I finally commit the changes

**Comment by tkoyama010** [comment]
Thanks, I was making that statement based on the fact that SciPy papers often emphasize being able to run the code and reproduce it. Do I need to run both Transformer and mamba to understand the results of this paper?

I think this paper could be an excellent introduction to mamba. To that end, I think it would be good to add a link to it appropriately (including Transformers) :)

**Comment by JaynouOliver** [comment]
Sure I can add those links in the paper. I will make these changes right away.

**Comment by JaynouOliver** [comment]
@tkoyama010 please add any further changes if needed after review.

File: papers/Suvrakamal_Das/mybib.bib [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  author    = {Buitinck, Lars and Louppe, Gilles and Blondel, Mathieu and Pedregosa, Fabian and Mueller, Andreas and Grisel, Olivier and Niculae, Vlad and Prettenhofer, Peter and Gramfort, Alexandre and Grobler, Jaques and Layton, Robert and VanderPlas, Jake and Joly, Arnaud and Holt, Brian and Varoquaux, Gaël},
+  booktitle = {ECML PKDD Workshop: Languages for Data Mining and Machine Learning},
+  year      = {2013},
+  pages     = {108--122},
+  title     = {{API} design for machine learning software: experiences from the scikit-learn project},
+}
+@MISC{mamba_github,
+  title       = "Mamba",
+  author      = "Albert Gu, Tri Dao",
+  url         = "https://github.com/state-spaces/mamba",
```
Comment (tkoyama010) [inline papers/Suvrakamal_Das/mybib.bib]: I'm sorry. In my operation, these URLs are not enabled in the Preview screen. Could you please check the operation to access it again, as I may have made a mistake?

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: papers/Suvrakamal_Das/mybib.bib (hunk 1 above)
Comment (JaynouOliver) [reply papers/Suvrakamal_Das/mybib.bib]: I am having problems with DOIs, how do I not mention a github link DOI and pass the checks? 
I tried removing the checks, but it isn't working well.
Either I have to try pasting a DOI for a github link, else I will have to remove the checks. 
Are there any other option?

**Comment by mepa** [comment]
Hi @tkoyama010 and @HaoZeke! In case a little extra time is needed, the initial complete review deadline has been extended to next [Wednesday, July 3rd](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#instructions-for-reviewers).

File: papers/Suvrakamal_Das/mybib.bib (hunk 1 above)
Comment (mepa) [reply papers/Suvrakamal_Das/mybib.bib]: Please see the sections "DOIs in bibliographies" and "Citing software and websites" in the [example paper](https://github.com/scipy-conference/scipy_proceedings/blob/2024/papers/00_myst_template/main.md) to see if that helps resolve the issue at hand.

**Comment by tkoyama010** [comment]
> Hi @tkoyama010 and @HaoZeke! In case a little extra time is needed, the initial complete review deadline has been extended to next [Wednesday, July 3rd](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#instructions-for-reviewers).

Yes please. Thanks!

File: papers/Suvrakamal_Das/mybib.bib (hunk 1 above)
Comment (JaynouOliver) [reply papers/Suvrakamal_Das/mybib.bib:335]: Yes, it is working, thank you so much, it will be solved today.

**Comment by JaynouOliver** [comment]
@tkoyama010 @mepa I see the github links are visible but not all checks are passsing due to missing DOI, if I add DOI (which doesn't exists for Github Links), the Github Links won't be visible. 
image

File: papers/Suvrakamal_Das/mybib.bib (hunk 1 above)
Comment (JaynouOliver) [reply papers/Suvrakamal_Das/mybib.bib]: Thanks for the references, we needed to add those keywords under myst.yml to pass the checks. It's not mentioned profoundly, but some other PR examples helped.

**Comment by JaynouOliver** [comment]
> @tkoyama010 please add any further changes if needed after review.

Hi @tkoyama010, all working and good!

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thanks!

File: papers/Suvrakamal_Das/main.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+**Recurrent Neural Networks (RNNs)** [@Sherstinsky_2020], often considered the natural choice for sequential data, are inherently stateful and require only constant computation per time step. However, they are slow to train and suffer from the well-known "**vanishing gradient problem**", which limits their ability to capture LRDs. **Convolutional Neural Networks (CNNs)** [@oshea2015introduction], while efficient for parallelizable training, are not inherently sequential and struggle with long context lengths, resulting in more expensive inference. **Transformers** [@vaswani2023attention], despite their recent success in various tasks, typically require specialized architectures and attention mechanisms to handle LRDs, which significantly increase computational complexity and memory usage.
```
Comment (HaoZeke) [inline papers/Suvrakamal_Das/main.md:25]: It isn't wholly clear from my reading of this, how the SSMs evade the vanishing gradients issue, which I thought had more to do with the training and, so, would show up possibly again with a large number of coefficients after HiPPO?

Comment (JaynouOliver) [reply papers/Suvrakamal_Das/main.md:25]: SSMs don't solve vanishing gradient problem but they have tricks to reduce the effect by HiPPO and NLPR, we have later discussed about this, we can refer to that section.

**Comment by JaynouOliver** [comment]
Hi everyone @tkoyama010  @HaoZeke  @ameyxd @mepa ,

I hope you’re all doing well. If there are any additional suggestions, feedback, or corrections needed before the author revision period ends on August 7th, please let me know. I’d be happy to make the necessary changes and incorporate your feedback.

Thank you very much!

**Review by tkoyama010 — APPROVED** [review]

**Comment by mepa** [comment]
Hi @tkoyama010 and @HaoZeke - Do you feel that this paper is ready for inclusion in the Proceedings?

@tkoyama010, I see that you have approved the PR so will assume "yes" unless I hear otherwise.

Thanks for reviewing!

**Comment by tkoyama010** [comment]
Yes! I am ready for it.

**Review by HaoZeke — APPROVED** [review]
LGTM, thanks for working on this!

**Comment by HaoZeke** [comment]
LGTM as well; go ahead :)

**Comment by JaynouOliver** [comment]
Thanks a lot @tkoyama010 and @HaoZeke for the approval!!

**Comment by mepa** [comment]
Thanks very much for reviewing, @tkoyama010 and @HaoZeke.

**Pull request merged**